Renamer/IconChanger.cs

```diff
@@ -0,0 +1,156 @@
+using System.Reflection;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.Events;
+using System.Linq;
+
+namespace Renamer
+{
+
+    [KSPAddon(KSPAddon.Startup.SpaceCentre, false)]
+    public class IconChanger_SpaceCentre : IconChanger
+    {
+        public void Awake()
+        {
+            GameEvents.onGUIAstronautComplexSpawn.Add(new EventVoid.OnEvent(OnGUIAstronautComplexSpawn));
+            GameEvents.onGUILaunchScreenVesselSelected.Add(new EventData<ShipTemplate>.OnEvent(OnGUILaunchScreenVesselSelected));
+        }
+    }
+
+    [KSPAddon(KSPAddon.Startup.EditorAny, false)]
+    public class IconChanger_EditorAny : IconChanger
+    {
+        public void Awake()
+        {
+            GameEvents.onEditorScreenChange.Add(new EventData<EditorScreen>.OnEvent(OnEditorScreenChange));
+        }
+    }
+
+    public class IconChanger : MonoBehaviour
+    {
+        public void OnGUIAstronautComplexSpawn()
+        {
+            StartCoroutine(CallbackUtil.DelayedCallback(1, BuildAstronautComplex));
+        }
+
+        public void OnGUILaunchScreenVesselSelected(ShipTemplate t)
+        {
+            StartCoroutine(CallbackUtil.DelayedCallback(1, BuildCrewAssignmentDialogue));
+        }
+
+        public void OnEditorScreenChange(EditorScreen e)
+        {
+            if (e == EditorScreen.Crew)
+            {
+                StartCoroutine(CallbackUtil.DelayedCallback(1, BuildCrewAssignmentDialogue));
+            }
+        }
+
+        public void BuildAstronautComplex()
+        {
+            ConfigNode data = null;
+            foreach (ConfigNode node in GameDatabase.Instance.GetConfigNodes("KERBALRENAMER"))
+            {
+                data = node;
+            }
+
+
+            KSP.UI.CrewListItem cic;
+            KSP.UI.UIList scroll;
+
+            UnityEngine.Object[] objs = Resources.FindObjectsOfTypeAll(typeof(KSP.UI.Screens.AstronautComplex));
+            if (objs.Length < 1) { return; }
+            KSP.UI.Screens.AstronautComplex complex = (KSP.UI.Screens.AstronautComplex)objs[0];
+            FieldInfo[] scrolls = typeof(KSP.UI.Screens.AstronautComplex).GetFields(BindingFlags.NonPublic | BindingFlags.Instance).Where(c => c.FieldType == typeof(KSP.UI.UIList)).ToArray();
+
+            for (int i = 0; i < scrolls.Length; i++)
+            {
+                scroll = (KSP.UI.UIList)scrolls[i].GetValue(complex);
+                for (int j = 0; j < scroll.Count; j++)
+                {
+                    KSP.UI.UIListItem listItem = scroll.GetUilistItemAt(j);
+                    cic = listItem.GetComponent<KSP.UI.CrewListItem>();
+                    cic.AddButtonInputDelegate(new UnityAction<KSP.UI.CrewListItem.ButtonTypes, KSP.UI.CrewListItem>(RebuildAstronautComplex));
+                    changeKerbalIcon(cic);
+                }
+            }
+        }
+
+        public void BuildCrewAssignmentDialogue()
+        {
+            if ((object)KSP.UI.CrewAssignmentDialog.Instance == null)
```

Comment:
I don't think you need to cast this to an object

Reply:
I removed all the unnecessary (object) casts